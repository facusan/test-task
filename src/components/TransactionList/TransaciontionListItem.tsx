import { Transaction } from "../../services/TransactionsService";
import {
  AvatarItemWrapper,
  DataWrapper,
  DescriptionWrapper,
  TitleWrapper,
  TransactionListItemWrapper,
} from "./TransactionListItem.styles";

export default function TransactionListItem({ date, value }: Transaction) {
  const getDateTimeFormat = (d: Date) =>
    d.getDate() +
    "-" +
    (d.getMonth() + 1) +
    "-" +
    d.getFullYear() +
    " " +
    d.getHours() +
    ":" +
    d.getMinutes();
  return (
    <TransactionListItemWrapper>
      <AvatarItemWrapper>
        <img style={{ height: "30px" }} alt="ether" src="ether.png"></img>
      </AvatarItemWrapper>
      <DataWrapper>
        <TitleWrapper>
          <div>{"Sent Ether"}</div>
          <div>{`-${value} ETH`}</div>
        </TitleWrapper>
        <DescriptionWrapper>
          <div>{getDateTimeFormat(date)}</div>
          <div>{`-$${value} USD`}</div>
        </DescriptionWrapper>
      </DataWrapper>
    </TransactionListItemWrapper>
  );
}
