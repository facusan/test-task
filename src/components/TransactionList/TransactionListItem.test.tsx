import renderer from "react-test-renderer";
import { Transaction } from "../../services/TransactionsService";
import TransaciontionListItem from "./TransaciontionListItem";

it("renders TransaciontionListItem correctly", () => {
  let transaction: Transaction = {
    date: new Date(0),
    from: "#123",
    id: 1,
    to: "#456",
    value: 100,
  };
  const component = renderer.create(
    <TransaciontionListItem {...transaction}></TransaciontionListItem>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
