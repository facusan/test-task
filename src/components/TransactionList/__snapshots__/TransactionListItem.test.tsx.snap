// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TransaciontionListItem correctly 1`] = `
<div
  className="sc-bczRLJ cbxpUU"
>
  <div
    className="sc-gsnTZi ctulah"
  >
    <img
      alt="ether"
      src="ether.png"
      style={
        Object {
          "height": "30px",
        }
      }
    />
  </div>
  <div
    className="sc-dkzDqf kEkHqq"
  >
    <div
      className="sc-hKMtZM fJHpc"
    >
      <div>
        Sent Ether
      </div>
      <div>
        -100 ETH
      </div>
    </div>
    <div
      className="sc-eCYdqJ edlkWG"
    >
      <div>
        27-9-2022 15:6
      </div>
      <div>
        -$100 USD
      </div>
    </div>
  </div>
</div>
`;
